### ISSUE TITLE
[BUG] Fix shim-related bugs 

### ISSUE THREAD

**Issue by gerashegalov** [NVIDIA/spark-rapids#5707]
**Describe the bug**
#5646 identified the following Shim loading bugs only visible if the AppClassloader is left unchanged (force.caller.classloader=false)

- [x] ExplainPlanImpl must be shimmed
- [x] ExecutionPlanCaptureCallback calls SparkShimImpl getSparkShimVersion unnecessarily

**Steps/Code to reproduce bug**
See #5646 

**Expected behavior**
force.caller.classloader=false should not throw NoClassDefFoundError

**Environment details (please complete the following information)**
 - any

**Additional context**
N/A

**Comment by tgravescs** [comment]
> ExplainPlanImpl must be shimmed

What does this mean exactly?  It is shimmed:

```
def explainPotentialGpuPlan(df: DataFrame, explain: String = "ALL"): String = {
    try {
      ShimLoader.newExplainPlan.explainPotentialGpuPlan(df, explain)

def newExplainPlan(): ExplainPlanBase = {
    newInstanceOf[ExplainPlanBase]("com.nvidia.spark.rapids.ExplainPlanImpl")
  }
```

**Comment by tgravescs** [comment]
nevermind, I think it means to say the com/nvidia/spark/rapids/ExplainPlan class needs to be in the unshimmmed classes so its user visible.

**Comment by gerashegalov** [comment]
Fixed by #5708
